clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1302,6 +1341,13 @@ public DescribeTopicsResult describeTopics(final Collection<String> topicNames,
             @Override
             void handleResponse(AbstractResponse abstractResponse) {
                 MetadataResponse response = (MetadataResponse) abstractResponse;
+                // Check for controller change
+                for (Errors error : response.errors().values()) {
+                    if (error == Errors.NOT_CONTROLLER) {
```

Comment:
Is the `NOT_CONTROLLER` error possible for Metadata requests? Also, are there any other errors we care about at this level (e.g. NOT_LEADER errors)?

Reply:
No, it's not.  Good catch.

"Not leader for partition" is something we care about for certain calls, but handling it will be harder.  I want to do that in a follow-on change.